### PR TITLE
#159987730 Users should be able to like or dislike comment made on an article.

### PR DESCRIPTION
**Pull request by emekafredy**
# What does this PR do?
Create endpoints for to like or dislike article comments

# Description of Task to be completed?
### Create endpoint
- Like comment ``` POST api/articles/<article_slug>/comments/<commentId>/like ```
- Dislike comment ``` POST api/articles/<article_slug>/comments/<commentId>/dislike ```

# How should this be manually tested?
- Clone the repo
- cd into the 'Thor-ah'
- Run 'npm install'
- Use the .env.sample format to create a .env file
- Run 'npm run migrate'
- Run 'npm run start:dev'

# Any background context you want to provide?
N/A

# What are the relevant pivotal tracker stories?
[159987730](https://www.pivotaltracker.com/story/show/159987730)

### CODE REVIEW

**Review by CEOehis — APPROVED** [review]
LGTM

**Review by lauragift21 — COMMENTED** [review]
LGTM

**Review by lauragift21 — CHANGES_REQUESTED** [review]
For naming your models, you can add underscore between the words making it comment_like.js.

**Review by lauragift21 — APPROVED** [review]
LGTM

**Comment by emekafredy** [comment]
@lauragift21 This has been resolved

File: server/controllers/comments.js [hunk 1]
```diff
@@ -122,6 +122,204 @@ class CommentsController {
...
+    if (req.params.reaction === 'like') {
+      return CommentsController.commentReactionQuery(req, res, 'like', 'disliked', '0', '1');
+    }
+    if (req.params.reaction === 'dislike') {
+      return CommentsController.commentReactionQuery(req, res, 'dislike', 'liked', '1', '0');
+    }
+  }
+
+  /**
+   * @description get number of likes and dislikes in an article
```
Comment (aknwosu) [inline server/controllers/comments.js]: misleading comment

File: server/controllers/comments.js [hunk 2]
```diff
@@ -122,6 +122,204 @@ class CommentsController {
...
+                  likes: 0,
+                  dislikes: 0,
+                }
+              });
+            }
+            commentLikes.forEach((commentReaction) => {
+              if (commentReaction.reaction === '1') {
+                commentReactions.likes += 1;
+              } else if (commentReaction.reaction === '0') {
+                commentReactions.dislikes += 1;
```
Comment (aknwosu) [inline server/controllers/comments.js]: a better implementation is to find all where reaction === 0 or reaction === 1 and then total number of likes would be the length of the returned value.

**Review by aknwosu — CHANGES_REQUESTED** [review]
add a function to check if the current logged in user already liked or disliked a comment so it can be used to format the view in the front end

File: server/controllers/comments.js (hunk 1 above)
Comment (emekafredy) [reply server/controllers/comments.js]: Thanks. I will correct this

File: server/controllers/comments.js (hunk 2 above)
Comment (emekafredy) [reply server/controllers/comments.js]: Thank you. I will also work on this

**Pull request merged**